stl/inc/shared_mutex

```diff
@@ -227,46 +279,82 @@ public:
 
     shared_lock() noexcept = default;
 
-    _NODISCARD_CTOR_LOCK explicit shared_lock(mutex_type& _Mtx)
+    _NODISCARD_CTOR_LOCK
+    _Acquires_shared_lock_(_Mtx)
+    _Acquires_nonreentrant_lock_(_Mtx)
+    _Acquires_shared_lock_(*this->_Pmtx)
+    _Acquires_nonreentrant_lock_(*this->_Pmtx)

```

Comment:
Is... is this necessary, given that we know `_Pmtx = &_Mtx` and lines 283/4 exist?

Reply:
Not necessary this way, I will replace with `_Post_same_lock_` .